pkg/controller/admissionchecks/provisioning/indexer.go

```diff
@@ -52,23 +54,20 @@ func indexWorkloadsChecks(obj client.Object) []string {
 	return slices.Map(wl.Status.AdmissionChecks, func(c *kueue.AdmissionCheckState) string { return c.Name })
 }
 
-func indexAdmissionCheckConnfig(obj client.Object) []string {
-	ac, isAc := obj.(*kueue.AdmissionCheck)
-	if !isAc || ac.Spec.ControllerName != ControllerName || !parametersRefValid(ac.Spec.Parameters) {
-		return nil
-	}
-
-	return []string{ac.Spec.Parameters.Name}
-}
-
-func SetupIndexer(ctx context.Context, indexer client.FieldIndexer) error {
+func SetupIndexer(ctx context.Context, indexer client.FieldIndexer, schema *runtime.Scheme) error {
 	if err := indexer.IndexField(ctx, &autoscaling.ProvisioningRequest{}, RequestsOwnedByWorkloadKey, indexRequestsOwner); err != nil {
 		return fmt.Errorf("setting index on provisionRequest owner: %w", err)
 	}
 	if err := indexer.IndexField(ctx, &kueue.Workload{}, WorkloadsWithAdmissionCheckKey, indexWorkloadsChecks); err != nil {
 		return fmt.Errorf("setting index on workloads checks: %w", err)
 	}
-	if err := indexer.IndexField(ctx, &kueue.AdmissionCheck{}, AdmissionCheckUsingConfigKey, indexAdmissionCheckConnfig); err != nil {
+
+	idxFnc, err := admissioncheck.GetIndexerByConfigFnc(ControllerName, &kueue.ProvisioningRequestConfig{}, schema)
```

Comment:
We pass the empty `ProvisioningRequestConfig` object just to retrieve its GVK. Can we pass the GVK directly?

Reply:
We may add in a future a more generic function if needed in the future, but the need seems questionable for now, since we can just pass the gvk, without going over the object.

Reply:
the more generic approach wold have helped, however done